Add WaitUntil / WaitWhile support to EditorCoroutine

hxdnshx opened the pull request.

hxdnshx committed:
Add WaitUntil / WaitWhile support to EditorCoroutine

hxdnshx commented:
In addition,it would be better to making the minimum version requirements of the Editor lower.
These codes work well in Unity 5.6.0.

marijnz commented:
Hey, sorry for the late response! I realised I don't get notifications for PR's and am now going through all of my repos.. If you're still up for it, there's two things:
1) WaitUntil and WaitWhile both implement CustomYieldInstruction! To support those and other ones, it's good enough to just check for that type and use "keepWaiting" instead. Also prevents having to use reflection.
2) The project is using tabs, not spaces for indentation. As you have spaces here and there, it looks weird on Github, so it's worth fixing. 

Good idea to support these classes, I like them and they're definitely an addition.

hxdnshx committed:
add .gitignore for external git tools.

hxdnshx committed:
modify WaitUntil/WaitWhile handler to CustomYieldInstruction handler.

hxdnshx commented:
Thanks for your reply. It is a great suggestion that implementing a handler for CustomYieldInstruction.
I have modified my code, and fixed tab/space problem.
@marijnz <- I hope Github works well about mention notification.

marijnz commented on .gitignore:
```diff
@@ -0,0 +1,36 @@
+[Ll]ibrary/
```
There is already a .gitignore, any reason to add another one?

hxdnshx replied:
git reads .gitignore rather than Unity.gitignore. I'll rename Unity.gitignore to the correct name.
Ref: https://git-scm.com/docs/gitignore

marijnz commented on Assets/EditorCoroutines/Scripts/EditorCoroutines.cs:
```diff
@@ -75,6 +75,20 @@ public bool IsDone(float deltaTime)
+                /*
+                 * To keep coroutine suspended return true. To let coroutine proceed with execution return false.
+                 * From: https://docs.unity3d.com/ScriptReference/CustomYieldInstruction.html
+                 */
```
Two small things here: 
1) The indentation is off, as it's spaces instead of tab. So it looks odd on GitHub. Better to fix that.
2) I think the comment here is not needed, the code is quite explicit in what it does. The comment only distracts for me.

marijnz commented on Assets/EditorCoroutines/Scripts/EditorCoroutines.cs:
```diff
@@ -354,6 +368,13 @@ static bool Process(EditorCoroutine coroutine)
+			{
+				coroutine.currentYield = new YieldCustomYieldInstruction()
+				{
+					customYield = current as CustomYieldInstruction
```
Perfect! 👍

marijnz reviewed (changes requested):
Good changes, but couple things still good to look at.

hxdnshx committed:
Fix indentation

hxdnshx committed:
rename gitignore

marijnz commented on Assets/EditorCoroutines/Scripts/CoroutineWindowExample.cs:
```diff
@@ -42,6 +42,28 @@ void OnGUI()
+			}
+		}
+
+		private bool _status;
```
Can this still be renamed to status? Last thing 👍

hxdnshx committed:
rename _status to status

marijnz commented:
Thanks! 👍
